xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/historyinline.vm

```diff
@@ -148,7 +148,7 @@ $xwiki.ssfx.use('uicomponents/pagination/pagination.css', true)##
     #pagination ($paginationParameters)
       <div class="form-inline">
         #if ($displayCompare)
-          <input type="submit" accesskey="c" value="$services.localization.render('core.viewers.history.compare')" class="btn btn-primary"/>
+          <input type="submit" accesskey="c" value="$services.localization.render('core.viewers.history.compare')" onclick="var rev1 = ''; var rev2 = ''; $$('[name=rev1], [name=rev2]').each(function(item) {if(item.getValue()) { if(item.name == 'rev1') rev1 = item.getValue(); else rev2 = item.getValue(); }});document.forms.${formname}.action='$tdoc.getURL('view', "viewer=changes&amp;rev1=__rev1__&amp;rev2=__rev2__&amp;$docvariant")'.replace('__rev1__', rev1).replace('__rev2__', rev2)" class="btn btn-primary"/>
```

Comment:
Is there a way to make this more readable and add comments to explain it? :)

Reply:
I think the clearer would be to put this JS in a dedicated file actually. If you look two line after my change there is exactly the same kind of JS: it would be easier to maintain if everything is put in one place and readable.
Is it fine if I create an history.js?

Reply:
Else I can also put directly the JS in a script in this vm

Reply:
Would be best to get feedback from @mflorea for example as I'm not an expert on our best practices related to JS ;)

Reply:
Looks better in a dedicated file. Then I need to call it with JSFX right? Or is there another way to include it?

Reply:
Yes, ``$xwiki.jsfx.use()``

Reply:
Thanks, I'll do that then.